dependency-management/dependency-management-impl/src/main/java/io/cloudslang/dependency/impl/services/DependencyServiceImpl.java

```
@@ -200,6 +233,37 @@ private void buildDependencyFile(String[] gav) {
             throw new IllegalStateException(fileToReturn.getPath() + " not found");
         }
         appendSelfToPathFile(gav, fileToReturn);
+        sendMavenDependencyBuildFinishedEvent(gav);
+    }
+
+    private void sendMavenDependencyBuildFinishedEvent(String[] gav) {
+        String message = "Artifact download complete";
+        if (gav.length >= 3) {
+            message = String.format("Downloading complete for artifact with gav: %s:%s:%s ", gav[0], gav[1], gav[2]);
+        }
+        Map<String, Serializable> data = new HashMap<>();
+        data.put(EventConstants.MAVEN_DEPENDENCY_BUILD_FINISHED, message);
+        ScoreEvent eventWrapper = new ScoreEvent(EventConstants.MAVEN_DEPENDENCY_BUILD_FINISHED, (Serializable) data);
+        dispatchEvent(eventWrapper);
+    }
+
+    private void sendMavenDependencyBuildEvent(String[] gav) {
+        String message = "Downloading artifact..";
+        if (gav.length >= 3) {
+            message = String.format("Downloading artifact with gav: %s:%s:%s ", gav[0], gav[1], gav[2]);
+        }
+        Map<String, Serializable> data = new HashMap<>();
+        data.put(EventConstants.MAVEN_DEPENDENCY_BUILD, message);
+        ScoreEvent eventWrapper = new ScoreEvent(EventConstants.MAVEN_DEPENDENCY_BUILD, (Serializable) data);
+        dispatchEvent(eventWrapper);
+    }
+
+    private void dispatchEvent(ScoreEvent eventWrapper) {
+        try {
+            eventBus.dispatch(eventWrapper);
+        } catch (InterruptedException e) {
+            logger.error("Thread is interrupted. Ignoring... ", e);

```

I think it is better to throw an exception at this point